File: 2_thread1.java
import java.text.SimpleDateFormat;
import java.util.Date;

public class Main {
    public static void main(String[] args) {
        outLog(new Object(){}.getClass().getEnclosingClass().getName());

        ThreadClsA threadA = new ThreadClsA();
        threadA.start();

        ThreadClsB threadB = new ThreadClsB();
        threadB.start();

        try {
            threadA.join();
            threadB.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    static class ThreadClsA extends Thread {
        public void run() {
            for (int i = 0; i < 2; i++) {
                try {
                    Thread.sleep(100);
                    outLog(new Object(){}.getClass().getEnclosingClass().getName());
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    static class ThreadClsB extends Thread {
        public void run() {
            for (int i = 0; i < 2; i++) {
                try {
                    Thread.sleep(150);
                    outLog(new Object(){}.getClass().getEnclosingClass().getName());
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    private static void outLog(String Cls) {
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss.SSS");
        String formatNowDate = sdf.format(new Date());
        System.out.println("CLS:" + Cls + " ID:" + Thread.currentThread().getId() + " Date:" + formatNowDate);
    }
}

File: 2_thread1.py
# coding: utf-8

import datetime
import os
import threading
import time

def out_log():
    # JST = datetime.timezone(datetime.timedelta(hours=+9), 'JST')
    # now = datetime.datetime.now(JST)
    now = datetime.datetime.now()
    print('Thread Name:{} Time:{}'.format(threading.currentThread().getName(), now))

def thread_funcA():
    for i in range(2):
        time.sleep(0.1)
        out_log()

def thread_funcB():
    for i in range(2):
        time.sleep(0.15)
        out_log()

def main():
    out_log()

    # 名前指定しなければ連番でThread-1のようにが自動命名される
    threadA = threading.Thread(target=thread_funcA)
    threadB = threading.Thread(target=thread_funcB)

    threadA.start()
    threadB.start()

    threadA.join()
    threadB.join()

    os._exit

if __name__ == "__main__":
    main()
